code/metrics.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+
+
+def hits_at_k(y_true, y_pred=None, k=None, is_l2=False):
+    """
+    Given distances between expected email representation and actual email representation,
+    or distribution of the probability that a sender receiver pair has sent a message
+    Hits @ k tells us that of the most confident k predictions, how many of them were correct
+    :param y_true: ndarray having shape (num_examples, ) indicating whether the mail was sent by the s-r pair
+    :param y_pred: ndarray having shape (num_examples, ) containing scores or probabilities
+    :param k: custom value for k; if this is not set, it considers the entire array
+    :param is_l2: flag to indicate whether the y_pred values are scores/errors (True) or probabilities (False)
+    :return: <int> the hits @ k metric
+    """
+    if k is None:
+        k = len(y_true)
+    else:
+        k = min(k, len(y_true))
+
+    if is_l2:
+        sort_order = y_pred.argsort()
+        y_true = y_true[sort_order]
+        correct = 0
+        for i in xrange(k):
+            correct += y_true[i]
+        return correct, k
```

Comment:
shouldn't we return a ratio e.g correct/k ?

Reply:
I believe both are equivalent but it seemed to me that we will not be saying that we are getting .8 hits @ K; instead say 4 hits @ 5